tests/unit/docs-lint-test.js

```diff
@@ -1,45 +1,61 @@
 'use strict';
 
+/*
+ * Turns out, YUIDoc attached a process.on('unhandledRejection' handler which
+ * incorrectly reports any unhandledRejection as a YUIDoc error. This makes
+ * debugging of unhandledRejections in the ember-cli test suite quite painful
+ *
+ * To address YUIDocs behavior, we run YUIDoc isolated in it's own process;
+ */
+if (!process.env['IS_CHILD_THREAD']) {
+  const execa = require('execa');
+
+  describe('YUIDoc', function() {
+    it('parses without warnings', async function() {
+      await execa('node', [`--unhandled-rejections=strict`, __filename], {
+        env: {
+          IS_CHILD_THREAD: true,
+        },
+      });
+    });
+  });
+  return;
+}
+
 const Y = require('yuidocjs');
 const EOL = require('os').EOL;
 
-describe('YUIDoc', function() {
-  let options = Y.Project.init({
-    quiet: true,
-  });
-  let yuiDoc = new Y.YUIDoc(options);
-
-  it('parses without warnings', function() {
-    let json = yuiDoc.run();
+let options = Y.Project.init({
+  quiet: true,
+});
+let yuiDoc = new Y.YUIDoc(options);
 
-    let warnings = {};
-    json.warnings.forEach(function(warning) {
-      let tmp = warning.line.split(':');
-      let file = tmp[0].trim();
-      let line = tmp[1];
+let json = yuiDoc.run();
 
-      if (!warnings[file]) {
-        warnings[file] = [];
-      }
+let warnings = {};
+json.warnings.forEach(function(warning) {
+  let tmp = warning.line.split(':');
+  let file = tmp[0].trim();
+  let line = tmp[1];
```

Comment:
Seems like a good place to destructure, and I'm looking for a chance to make a multi-line suggestion

```suggestion
  let [file, line] = warning.line.split(':').map(item => item.trim());
```

Reply:
Ya, that's a fine suggestion. But in general I try to avoid cosmetic cleanup in bug-fixes.